### PR TITLE
ref: For envelopeWithData serialization

### PR DESCRIPTION
**Pull request by philipphofmann**
Small refactorings for SentrySerialization.envelopeWithData.

This PR is based on https://github.com/getsentry/sentry-cocoa/pull/4342.

#skip-changelog

### CODE REVIEW

File: Sources/Sentry/SentrySerialization.m [hunk 1]
```diff
@@ -100,32 +101,33 @@ + (SentryEnvelope *_Nullable)envelopeWithData:(NSData *)data
...
-
-                envelopeHeader = [[SentryEnvelopeHeader alloc] initWithId:eventId
-                                                                  sdkInfo:sdkInfo
-                                                             traceContext:traceContext];
-
-                if (headerDictionary[@"sent_at"] != nil) {
-                    envelopeHeader.sentAt = sentry_fromIso8601String(headerDictionary[@"sent_at"]);
-                }
+                break;
+            }
```
Comment (philipphofmann) [inline Sources/Sentry/SentrySerialization.m:105]: I added the break here to reduce the code indentation. No changes to the code itself.

File: Sources/Sentry/SentrySerialization.m [hunk 2]
```diff
@@ -100,32 +101,33 @@ + (SentryEnvelope *_Nullable)envelopeWithData:(NSData *)data
...
+
+            envelopeHeader = [[SentryEnvelopeHeader alloc] initWithId:eventId
+                                                              sdkInfo:sdkInfo
+                                                         traceContext:traceContext];
+
+            if (headerDictionary[@"sent_at"] != nil) {
+                envelopeHeader.sentAt = sentry_fromIso8601String(headerDictionary[@"sent_at"]);
+            }
+
             break;
```
Comment (brustolin) [inline Sources/Sentry/SentrySerialization.m:131]: l: What is this break at the end of the loop?

**Review by brustolin — APPROVED** [review]

File: Sources/Sentry/SentrySerialization.m (hunk 2 above)
Comment (philipphofmann) [reply Sources/Sentry/SentrySerialization.m:130]: It's not at the end of the loop. It's at the end of the if statement. It also confused me a bit when I looked at it the first time 😁

Comment (brustolin) [reply Sources/Sentry/SentrySerialization.m:130]: But after the If the loops ends, what I'm missing here?

Comment (brustolin) [reply Sources/Sentry/SentrySerialization.m:130]: Forget about it. I got it now. I was thinking about `continue`

**Pull request merged**